File: Plugins/AM_PRIDE_Converter_PlugIn/clsPXFileInfoBase.vb
Imports AnalysisManagerBase.clsAnalysisResources

Public Class clsPXFileInfoBase

#Region "Module Variables"
    Protected mFileName As String
    Protected mJobInfo As udtDataPackageJobInfoType
#End Region

#Region "Structures and Enums"
    Public Enum ePXFileType
        Undefined = 0
        Result = 1              ' .msgf-pride.xml files
        ResultMzId = 2          ' .mzid files from MSGF+  (listed as "result" files in the .px file)
        Raw = 3                 ' Instrument data files (typically .raw files)
        Search = 4              ' Search engine output files, such as Mascot DAT or other output files (from analysis pipelines, such as pep.xml or prot.xml).
        Peak = 5                ' _dta.txt or .mgf files
    End Enum
#End Region

#Region "Auto-properties"
    Public Property FileID As Integer
    Public Property Length As Int64
    Public Property MD5Hash As String
#End Region

#Region "Properties"
    Public ReadOnly Property Filename() As String
        Get
            Return mFileName
        End Get
    End Property

    Public ReadOnly Property JobInfo() As udtDataPackageJobInfoType
        Get
            Return mJobInfo
        End Get
    End Property
#End Region

    Public Sub New(fileName As String, udtJobInfo As udtDataPackageJobInfoType)
        mFileName = fileName
        mJobInfo = udtJobInfo
    End Sub

    Public Sub Update(oSource As clsPXFileInfoBase)
        Me.mFileName = oSource.mFileName
        Me.mJobInfo = oSource.mJobInfo
        Me.FileID = oSource.FileID
        Me.Length = oSource.Length
        Me.MD5Hash = oSource.MD5Hash
    End Sub

    Public Overrides Function ToString() As String
        Return "Job " & mJobInfo.Job & ": " & mFileName
    End Function

End Class

File: Plugins/AM_PRIDE_Converter_PlugIn/clsPXFileInfoBase.cs
using AnalysisManagerBase;

namespace AnalysisManagerPRIDEConverterPlugIn
{
    /// <summary>
    /// ProteomeXchange file info base class
    /// </summary>
    public class clsPXFileInfoBase
    {

        #region "Structures and Enums"

        /// <summary>
        /// ProteomeXhcnage file type
        /// </summary>
        public enum ePXFileType
        {
            /// <summary>
            /// Undefined
            /// </summary>
            Undefined = 0,

            /// <summary>
            /// .msgf-pride.xml files
            /// </summary>
            Result = 1,

            /// <summary>
            /// .mzid.gz files from MSGF+  (listed as "result" files in the .px file)
            /// </summary>
            ResultMzId = 2,

            /// <summary>
            /// Instrument data files (typically .raw files)
            /// </summary>
            Raw = 3,

            /// <summary>
            /// Search engine output files, such as Mascot DAT or other output files (from analysis pipelines, such as pep.xml or prot.xml).
            /// </summary>
            Search = 4,

            /// <summary>
            /// _dta.txt or .mgf files
            /// </summary>
            Peak = 5
        }

        #endregion

        #region "Auto-properties"

        /// <summary>
        /// File ID
        /// </summary>
        public int FileID { get; set; }

        /// <summary>
        /// File size (in bytes)
        /// </summary>
        public long Length { get; set; }

        /// <summary>
        /// MD5 hash
        /// </summary>
        public string MD5Hash { get; set; }

        #endregion

        #region "Properties"

        /// <summary>
        /// Filename
        /// </summary>
        public string Filename { get; private set; }

        /// <summary>
        /// Job Info
        /// </summary>
        public clsDataPackageJobInfo JobInfo { get; private set; }

        #endregion

        /// <summary>
        /// Constructor
        /// </summary>
        /// <param name="fileName"></param>
        /// <param name="dataPkgJob"></param>
        public clsPXFileInfoBase(string fileName, clsDataPackageJobInfo dataPkgJob)
        {
            Filename = fileName;
            JobInfo = dataPkgJob;
        }

        /// <summary>
        /// Update Filename, JobInfo, FileID, Length, and MD5 hash
        /// </summary>
        /// <param name="oSource"></param>
        public void Update(clsPXFileInfoBase oSource)
        {
            Filename = oSource.Filename;
            JobInfo = oSource.JobInfo;
            FileID = oSource.FileID;
            Length = oSource.Length;
            MD5Hash = oSource.MD5Hash;
        }

        /// <summary>
        /// Returns Job number and filename
        /// </summary>
        /// <returns></returns>
        public override string ToString()
        {
            return "Job " + JobInfo.Job + ": " + Filename;
        }
    }
}
